chanzuckerberg/single-cell-data-portal
# feat(wmg): Newsletter Signup

## Reason for Change

- #4453 

Readability: @prathapsridharan 
Functionality: @atarashansky 

## Changes

- add
    - Adding new Newsletter component
- remove
- modify

## Testing steps

rdev: https://newsletter-signup-frontend.rdev.single-cell.czi.technology/gene-expression
newsletter direct link: https://newsletter-signup-frontend.rdev.single-cell.czi.technology/gene-expression?newsletter_signup=true

1. Navigate to gene expression, datasets, or collections page
2. Click "subscribe" button in the banner to open the modal and ensure modal opens with content and logo
3. Enter an invalid and/or valid email address in the email input box
4. Press "subscribe" button
5. If an invalid email was entered, you should see an error message underneath the input box
6. If a valid email was entered, ensure you see "Thanks for subscribing!" text after submission
7. Click the X button to close out the modal
7. Refresh the page to reset the newsletter banner to input an email again

Home landing page
1. Scroll to the bottom of the page
2. Ensure banner is attached to the footer of the page (note there is no modal to open on the home page)
3. Enter an invalid/valid email address
4. Press "submit" button
5. If an invalid email was entered, you should see an error message underneath the input box
6. If a valid email was entered, you should see "Thanks for subscribing!" text after submission
7. Refresh the page to reset the newsletter banner to input an email again

## Notes for Reviewer

Implementation is almost exactly how Napari Hub integrates with Hubspot in order to prevent saving cookies. Their PR for integrating with Hubspot can be viewed here: https://github.com/chanzuckerberg/napari-hub/pull/661/files
